### PR TITLE
Fix inconsistent formatting of queries

### LINKED ISSUE #52349: Broken AST formatting of select with intersect and union all
When i try to create a table in all cluster using a query with [intersect] and [union all], the table is created but the result is not as i expected. It seems like bracket does not work or wrong priority with [intersect] and [union all] in [create table on cluster as]

The select sql :
`select 1 as c
intersect (select 1 as c
            union all 
           select 2 as c
);`
image

use create table like this:
`create table ltv.t1 on cluster cluster_ltv_replacingmergetree engine=MergeTree() order by c as (
select 1 as c
intersect (select 1 as c
            union all 
           select 2 as c
          )
);
`
`select * from ltv.t1;`
result is 1 and 2

image

I have tried two versions:23.6.2.18 ,22.2.2.1

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Bug Fix

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix inconsistent formatting of queries containing tables named `table`. Fix wrong formatting of queries with `UNION ALL`, `INTERSECT`, and `EXCEPT` when their structure wasn't linear. This closes #52349. Fix wrong formatting of `SYSTEM` queries, including `SYSTEM ... DROP FILESYSTEM CACHE`, `SYSTEM ... REFRESH/START/STOP/CANCEL/TEST VIEW`, `SYSTEM ENABLE/DISABLE FAILPOINT`. Fix formatting of parameterized DDL queries. Fix the formatting of the `DESCRIBE FILESYSTEM CACHE` query. Fix incorrect formatting of the `SET param_...` (a query setting a parameter). Fix incorrect formatting of `CREATE INDEX` queries. Fix inconsistent formatting of `CREATE USER` and similar queries. Fix inconsistent formatting of `CREATE SETTINGS PROFILE`. Fix incorrect formatting of `ALTER ... MODIFY REFRESH`. Fix inconsistent formatting of window functions if frame offsets were expressions. Fix inconsistent formatting of `RESPECT NULLS` and `IGNORE NULLS` if they were used after a function that implements an operator (such as `plus`). Fix idiotic formatting of `SYSTEM SYNC REPLICA ... LIGHTWEIGHT FROM ...`. Fix inconsistent formatting of invalid queries with `GROUP BY GROUPING SETS ... WITH ROLLUP/CUBE/TOTALS`. Fix inconsistent formatting of `GRANT CURRENT GRANTS`. Fix inconsistent formatting of `CREATE TABLE (... COLLATE)`. Additionally, I fixed the incorrect formatting of `EXPLAIN` in subqueries (#60102). Fixed incorrect formatting of lambda functions (#60012). Added a check so there is no way to miss these abominations in the future.

See https://s3.amazonaws.com/clickhouse-test-reports/0/25f2501c7f643c656da796f86848dbfc5b509289/stateless_tests__debug__s3_storage_.html

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 69dfd721ceae315bc62ecb7145cebce3b2a58166 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60095/69dfd721ceae315bc62ecb7145cebce3b2a58166/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60095/69dfd721ceae315bc62ecb7145cebce3b2a58166/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by yakov-olkhovskiy** [comment]
12 tests using `table` name needs to be fixed
also end of line in the new test's reference file

**Comment by alexey-milovidov** [comment]
The lack of new line is legit :)

**Review by yakov-olkhovskiy — APPROVED** [review]
unit tests and some integration tests are failing also for the same reason...

**Comment by alexey-milovidov** [comment]
Also, check the place where `length_error` happens. If there is an overflow, we should also check for it.
I assume, it can gracefully return error - either something like "cannot allocate a thread" or "memory limit exceeded" - let's check it on multiple levels.

File: src/Parsers/ASTSystemQuery.cpp [hunk 1]
```diff
@@ -144,185 +152,273 @@ void ASTSystemQuery::formatImpl(const FormatSettings & settings, FormatState &,
...
+
+                // If the mode is LIGHTWEIGHT and specific source replicas are specified
+                if (sync_replica_mode == SyncReplicaMode::LIGHTWEIGHT && !src_replicas.empty())
+                {
+                    settings.ostr << ' ';
+                    print_keyword("FROM");
+                    settings.ostr << ' ';
+
+                    bool first = true;
+                    for (auto & src : src_replicas)
```
Comment (yakov-olkhovskiy) [inline src/Parsers/ASTSystemQuery.cpp:232]: clang tidy
```suggestion
                    for (const auto & src : src_replicas)
```

**Comment by alexey-milovidov** [comment]
Wow, all the tests passed! Now only conflicts remain.

**Comment by clickgapai** [comment]
Hi @alexey-milovidov @yakov-olkhovskiy — while reviewing this PR I found the following:

- **Coverage gap:** `ASTSystemQuery` formatter: multi-replica `LIGHTWEIGHT FROM`, `KEY ... OFFSET`, and `DROP REPLICA FROM SHARD` branches lack format-roundtrip coverage — https://github.com/ClickHouse/ClickHouse/pull/103865

Could you review the test PR and add the `can be tested` label if it looks good?
Happy to discuss — close anything that's wrong or already addressed.

**Pull request merged**